src/psyclone/psyGen.py

```diff
@@ -5963,6 +6052,17 @@ def symbol_table(self):
         '''
         return self._symbol_table
 
+    def set_opencl_options(self, options):
+        '''
+        Set the opencl_options map. Note that KernelSchedule expects to
+        receive options which are already vaildated by the caller.
```

Comment:
I can see where you're coming from but I think it's better to be defensive and do the checks here too. Having said that, it would be even better if we only kept these properties in one place.

Reply:
Removed this method as agreed in the conversation below.